zebra-network/src/peer_set/initialize.rs

```diff
@@ -194,15 +194,28 @@ where
     S::Future: Send + 'static,
 {
     info!(?initial_peers, "Connecting to initial peer set");
-    use tower::util::CallAllUnordered;
-    let addr_stream = futures::stream::iter(initial_peers.into_iter());
-    let mut handshakes = CallAllUnordered::new(connector, addr_stream);
-
-    while let Some(handshake_result) = handshakes.next().await {
-        tx.send(handshake_result).await?;
+    use futures::TryStreamExt;
+
+    let fut = initial_peers
+        .clone()
+        .into_iter()
+        .map(|request| connector.clone().oneshot(request))
+        .collect::<futures::stream::FuturesUnordered<_>>()
+        .try_collect::<Vec<_>>();
```

Comment:
We don't want to use `try_collect` here:
> If an error happens then all collected elements will be dropped and the error will be returned.
https://docs.rs/futures/0.3.12/futures/stream/trait.TryStreamExt.html#method.try_collect

Reply:
You can keep most of the existing code if you just drop the `try_collect`.